src/elm/Page/Dashboard/Analysis.elm

```diff
@@ -498,29 +574,105 @@ update msg model loggedIn =
+
+                _ ->
+                    UR.init model
+
+        ClearSelectSelection ->
+            case model.status of
+                Loaded _ _ ->
+                    let
+                        oldFilters =
+                            model.filters
+
+                        newModel =
+                            { model
+                                | status = Loading
+                                , filters = { oldFilters | profile = Nothing }
+                                , reloadOnNextQuery = True
+                            }
+                    in
+                    newModel
+                        |> UR.init
+                        |> UR.addCmd (fetchAnalysis loggedIn newModel.filters Nothing)
 
                 _ ->
                     UR.init model
 
+        SelectStatusFilter statusFilter ->
+            let
+                oldFilters =
+                    model.filters
 
-fetchAnalysis : Shared -> Symbol -> Eos.Name -> Maybe String -> Cmd Msg
-fetchAnalysis shared communityId account maybeCursorAfter =
+                newModel =
+                    { model
+                        | filters = { oldFilters | statusFilter = statusFilter }
+                        , status = Loading
+                        , reloadOnNextQuery = True
+                    }
+            in
+            newModel
+                |> UR.init
+                |> UR.addCmd (fetchAnalysis loggedIn newModel.filters Nothing)
+
+
+fetchAnalysis : LoggedIn.Model -> Filter -> Maybe String -> Cmd Msg
+fetchAnalysis { accountName, selectedCommunity, shared } { profile, statusFilter } maybeCursorAfter =
     let
+        filterRecord =
+            { claimer = Absent
+            , status = Absent
+            }
+                |> (\fr ->
+                        { fr
+                            | claimer =
+                                case profile of
+                                    Just p ->
+                                        Present (Eos.nameToString p.account)
+
+                                    Nothing ->
+                                        Absent
+                        }
+                   )
+                |> (\fr ->
+                        { fr
+                            | status =
+                                case statusFilter of
+                                    All ->
+                                        Absent
+
+                                    Approved ->
+                                        Present "approved"
+
+                                    Rejected ->
+                                        Present "rejected"
+
+                                    Pending ->
+                                        Present "pending"
+                        }
+                   )
+
+        filter =
+            if filterRecord.claimer /= Absent || filterRecord.status /= Absent then
+                Present filterRecord
+
+            else
+                Absent
```

Comment:
Maybe it's better to use pattern-matching here? If I understood everything right, it will be something like this:

```elm
filter =
    case ( filterRecord.claimer, filterRecord.status ) of
        ( Absent, Absent ) ->
            Absent

        ( _, _ ) ->
            Present filterRecord
```

Reply:
nice!! I will apply it!